# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build gst-plugins-base-build"

PKG_CONFIG_PATH=/fb/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

if pkg-config --exists a52dec-build; then
  A52DEC=1
  REQUIRES="$REQUIRES a52dec-build"
fi

if pkg-config --exists lame-build; then
  LAME=1
  REQUIRES="$REQUIRES lame-build"
fi

if pkg-config --exists libmad-build; then
  MAD=1
  REQUIRES="$REQUIRES libmad-build"
fi

if pkg-config --exists x264-build; then
  X264=1
  REQUIRES="$REQUIRES x264-build"
fi

# configure (ac-2.69; am-1.15; lt-2.4.2)
libtoolize -c -f; aclocal -I m4 -I common/m4; autoheader; autoconf; automake -a -c
PKG_CONFIG_PATH=/fb/lib/pkgconfig LDFLAGS=-Wl,-rpath,/fb/lib ./configure --prefix=/fb

# build
make

# install
install -d $DESTDIR/fb/lib/gstreamer-1.0
test $A52DEC && install ext/a52dec/.libs/libgsta52dec.so $DESTDIR/fb/lib/gstreamer-1.0
test $LAME && install ext/lame/.libs/libgstlame.so $DESTDIR/fb/lib/gstreamer-1.0
test $MAD && install ext/mad/.libs/libgstmad.so $DESTDIR/fb/lib/gstreamer-1.0
test $X264 && install ext/x264/.libs/libgstx264.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/asfdemux/.libs/libgstasf.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/dvdlpcmdec/.libs/libgstdvdlpcmdec.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/dvdsub/.libs/libgstdvdsub.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/realmedia/.libs/libgstrmdemux.so $DESTDIR/fb/lib/gstreamer-1.0
install gst/xingmux/.libs/libgstxingmux.so $DESTDIR/fb/lib/gstreamer-1.0

# build.pc
install -d $DESTDIR/fb/share/pkgconfig
cat > $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
Name: gst-plugins-ugly
Version: 1.6.4
Description: Ugly GStreamer plugins
Requires: $REQUIRES

exec=\\
test $A52DEC && cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgsta52dec.so \\
EOF
cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstasf.so \\
/fb/lib/gstreamer-1.0/libgstdvdlpcmdec.so \\
/fb/lib/gstreamer-1.0/libgstdvdsub.so \\
EOF
test $LAME && cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstlame.so \\
EOF
test $MAD && cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstmad.so \\
EOF
cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstrmdemux.so \\
EOF
test $X264 && cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstx264.so \\
EOF
cat >> $DESTDIR/fb/share/pkgconfig/gst-plugins-ugly-build.pc << EOF
/fb/lib/gstreamer-1.0/libgstxingmux.so
EOF

exit

# patch
--- gst-plugins-ugly-1.6.4.orig/configure.ac
+++ gst-plugins-ugly-1.6.4/configure.ac
@@ -25,7 +25,7 @@
 AC_CONFIG_HEADERS(config.h)
 
 dnl AM_MAINTAINER_MODE only provides the option to configure to enable it
-AM_MAINTAINER_MODE([enable])
+AM_MAINTAINER_MODE
 
 dnl sets host_* variables
 AC_CANONICAL_HOST

# source
https://gstreamer.freedesktop.org/src/gst-plugins-ugly/gst-plugins-ugly-1.6.4.tar.xz
